### PR TITLE
Readded permissions to Role#edit's payload

### LINKED ISSUE #1759: role.setPermissions or role.edit doesn't work
**Please describe the problem you are having in as much detail as possible:**
I updated to master, and my working bot that creates a role and updates its permissions doesn't do that anymore.

I can get
```js
message.guild.createRole({
    data: {
        name: 'USER-' + message.author.id,
        color: args[0],
        permissions: []
    },
    reason: 'Hex Bot'
})
```
To create a role with no permissions, but if I put a .setPermissions or .edit in the .then it will create the role with two permissions it wasn't told to give. (External emojis and add reaction)

And trying to run
```js
role.edit({
    data: {
        color: args[0],
        permissions: ['ADMINISTRATOR']
    },
    reason: 'Hex Bot'
}).catch(error => client.catch(error))
```
Doesn't edit the permissions. same with setPermissions, it works in stable though.
And it does update the color.

**Further details:**

- discord.js version: Master (12.0.0-dev #317a352)
- node.js version: v8.2.1
- Operating system: Windows 10 x64 & Generic Linux x86_64
- Priority this issue should have – please be realistic and elaborate if possible:
Medium priority, Needs to be fixed before v12 is released but its still not a big problem for in-dev

- [x] I found this issue while running code on a __user account__
- [x] I found this issue while running code on a __bot account__
- [x] I have also tested the issue on latest master, commit hash: #317a352

I found this issue last night, and waited until this morning to update Discord.JS and try again, it still happens.

I have tried on both bot and user accounts. on three separate dev servers.

### PR DESCRIPTION
**Pull request by SpaceEEC**
**Please describe the changes this PR makes and why it should be merged:**

Fixes #1759 
The permissions property was missing in the data when patching a role.
Also sorted them the same way as the typedef and docs.

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by PgBiel** [comment]
Blazing fast...

File: src/structures/Role.js [hunk 1]
```diff
@@ -205,9 +205,10 @@ class Role {
     return this.client.api.guilds[this.guild.id].roles[this.id].patch({
       data: {
         name: data.name || this.name,
```
Comment (PgBiel) [inline src/structures/Role.js]: If you are re-ordering it alphabetically, isn't name supposed to be before `position`?

File: src/structures/Role.js [hunk 2]
```diff
@@ -205,9 +205,10 @@ class Role {
...
-        position: typeof data.position !== 'undefined' ? data.position : this.position,
         color: Util.resolveColor(data.color || this.color),
         hoist: typeof data.hoist !== 'undefined' ? data.hoist : this.hoist,
+        position: typeof data.position !== 'undefined' ? data.position : this.position,
+        permissions: data.permissions,
         mentionable: typeof data.mentionable !== 'undefined' ? data.mentionable : this.mentionable,
```
Comment (PgBiel) [inline src/structures/Role.js]: and this before `name`

Comment (SpaceEEC) [inline src/structures/Role.js]: Oh, I should have seen that.
Then just how the typedef and docs lists them.

**Comment by ShayBox** [comment]
Someone merge this please, My bot requires this to function.

**Comment by iCrawl** [comment]
@ShayBox this is not how this works here, and if you desperately need it, you can install the latest commit of this PR and run on it.

File: src/structures/Role.js (hunk 2 above)
Comment (PgBiel) [inline src/structures/Role.js]: okay

**Pull request merged**